builder/azure/common/client/config.go

```diff
@@ -295,6 +312,16 @@ func (c *Config) FillParameters() error {
 		}
 	}
 
+	if c.authType == authTypeAzureCLI && c.TenantID == "" {
```

Comment:
Do we have a service principal in the acceptance tests? Where are they configured?

Reply:
@nywilken I've removed `&& c.TenantID == ""` from here since we should always grab the tenantid/subscription id from the active session. No good response around the acceptance test yet, will look into it as well after fixing the low hanging fruit.

Reply:
I've done my best with an acceptance test (right now only for Managed Disk, but please tell me if you want me to add more). You can see the commit here: https://github.com/hashicorp/packer/pull/10157/commits/726439295bca6c8e4fa8b870451d41dd02ecf879

I was able to run the test locally:
```shell
export ARM_CLIENT_ID="<client id>"
export ARM_CLIENT_SECRET="<client secret>"
export ARM_TENANT_ID="<tenant id>"
PACKER_ACC=1 go test -v ./builder/azure/arm/ -run TestBuilderAcc_ManagedDisk_Linux_AzureCLI -timeout=20m
```

This resulted in the following:
```shell
--- PASS: TestBuilderAcc_ManagedDisk_Linux_AzureCLI (542.77s)
PASS
ok      github.com/hashicorp/packer/builder/azure/arm   543.282s
```

I'm running `az login --service-principal --username <client id> --password <client secret> --tenant <tenant id>` inside of the test. This required me to add some structs and some helper functions which I'm not sure you want in there.

The first thing we need to verify is that `ARM_TENANT_ID` is available during the acceptance tests and if it isn't I need some pointer on how to add it.

Looking forward to your feedback!